code/modules/research/protolathe.dm

```diff
@@ -106,6 +105,8 @@ Note: Must be placed west/left of and R&D console to function.
 		if(!in_range(src, stack) || !user.Adjacent(src))
 			return
 		var/amount_inserted = materials.insert_stack(O,amount)
+		if (amount_inserted <= 0)

```

Comment:
I can't reproduce it with the steps you described, it fails silently if I try to insert a negative amount of sheets.